python/metatensor_torch/metatensor/torch/atomistic/ase_calculator.py

```diff
@@ -501,31 +527,96 @@ def compute_energy(
+                # split them into the original systems
+                split_sizes = [len(system) for system in systems]
+                split_indices = np.cumsum(split_sizes[:-1])
+                results_as_numpy_arrays["forces"] = np.split(
+                    results_as_numpy_arrays["forces"], split_indices, axis=0
+                )
+
+                if all(atoms.pbc.all() for atoms in atoms_list):
+                    results_as_numpy_arrays["stress"] = [
+                        s
+                        for s in predictions["non_conservative_stress"]
+                        .block()
+                        .values.squeeze(-1)
+                        .detach()
+                        .cpu()
+                        .numpy()
+                    ]
+            else:
+                energy_tensor = energies.block().values
+                energy_tensor.backward(torch.ones_like(energy_tensor))
+                results_as_numpy_arrays["forces"] = [
+                    -system.positions.grad.cpu().numpy() for system in systems
                 ]
+                if all(atoms.pbc.all() for atoms in atoms_list):
+                    results_as_numpy_arrays["stress"] = [
+                        strain.grad.cpu().numpy() / atoms.cell.volume
+                        for strain, atoms in zip(strains, atoms_list)
+                    ]
         if was_single:
             for key, value in results_as_numpy_arrays.items():
                 results_as_numpy_arrays[key] = value[0]
         return results_as_numpy_arrays
 
+    def _ase_properties_to_metatensor_outputs(self, properties):
+        energy_properties = []
+        for p in properties:
+            if p in ["energy", "energies", "forces", "stress", "stresses"]:
+                energy_properties.append(p)
+            else:
+                raise PropertyNotImplementedError(
+                    f"property '{p}' it not yet supported by this calculator, "
+                    "even if it might be supported by the model"
+                )
+
+        output = ModelOutput()
+        output.quantity = "energy"
+        output.unit = "ev"
+        output.explicit_gradients = []
+
+        if "energies" in properties or "stresses" in properties:
+            output.per_atom = True
+        else:
+            output.per_atom = False
+
+        if "stresses" in properties:
+            output.explicit_gradients = ["cell"]
+
+        metatensor_outputs = {"energy": output}
+        if "forces" in properties and self._non_conservative:
+            metatensor_outputs["non_conservative_forces"] = ModelOutput(
+                quantity="forces", unit="eV/Angstrom", per_atom=True
+            )
+        if "stress" in properties and self._non_conservative:
+            metatensor_outputs["non_conservative_stress"] = ModelOutput(
+                quantity="stress", unit="eV/Angstrom^3", per_atom=False
+            )
```

Comment:
We should check that the model can actually do these before asking it to

Reply:
Now checked in the init

Reply:
Since the outputs we need could change from one step to another, I would rather do the check at runtime. I'll do a change myself